vertigo-bot-designer/src/main/java/io/vertigo/chatbot/designer/builder/services/TrainingServices.java

```diff
@@ -148,15 +152,31 @@ public Training trainAgent(@SecuredOperation("botContributor") final Chatbot bot
 					});
 			LogsUtils.addLogs(logs, "Call training OK, training in progress...");
 			return training;
+
 		} catch (final Exception e) {
 			LogsUtils.logKO(logs);
 			LogsUtils.addLogs(logs, e.getMessage());
 			LOGGER.error("error", e);
 			training.setLog(logs.toString());
-			throw new VSystemException("error training", e);
-		} finally {

```

Comment:
plus besoin du finally ?

Reply:
Il faut que je renvoie le training dans tous les cas, et en fait eclipse me met un warning "finally block does not complete normally" quand j'essaie de lui mettre les lignes suivantes :
training.setLog(logs.toString());
saveTraining(bot, training);
return training;

Donc je le fait à la fois dans le cours normal et dans le catch finalement

Reply:
(mais yavait bien une petite erreur du coup, je corrige)

Reply:
c'est a cause de ton return dans le finally. Le finally sert à faire un clean de certains objets. Donc tu peux sortir ton return et tu laisses le bloc finally